.github/check-format.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+source $(dirname "${BASH_SOURCE[0]}")/bot-base.sh
+
+echo -n "Collecting information on triggering PR"
+PR_URL=$(jq -r .pull_request.url "$GITHUB_EVENT_PATH")
+if [[ "$PR_URL" == "null" ]]; then
+  echo -n ........
+  PR_URL=$(jq -er .issue.pull_request.url "$GITHUB_EVENT_PATH")
+  echo -n .
+fi
+echo -n .
+PR_JSON=$(api_get $PR_URL)
+echo -n .
+ISSUE_URL=$(echo "$PR_JSON" | jq -er ".issue_url")
+echo -n .
+HEAD_REPO=$(echo "$PR_JSON" | jq -er .head.repo.full_name)
+echo -n .
+HEAD_BRANCH=$(echo "$PR_JSON" | jq -er .head.ref)
+echo .
+HEAD_URL="https://${GITHUB_ACTOR}:${GITHUB_TOKEN}@github.com/$HEAD_REPO"
+
+bot_comment() {
+  api_post "$ISSUE_URL/comments" "{\"body\":\"$1\"}" > /dev/null
+}
+
+bot_error() {
+  echo "$1"
+  bot_comment "Error: $1"
+  exit 1
+}
+
+# save scripts from develop
+cd dev_tools/scripts
+cp add_license.sh update_ginkgo_header.sh format_header.sh ../../../
+cd ../../
+
+git remote add fork "${HEAD_URL}"
+git fetch fork "$HEAD_BRANCH"
+git checkout -b format-tmp-$HEAD_BRANCH "fork/$HEAD_BRANCH"
+
+# restore files from develop
+cp ../add_license.sh dev_tools/scripts/
+cp ../update_ginkgo_header.sh dev_tools/scripts/
+cp ../format_header.sh dev_tools/scripts/
+
+# format files
+dev_tools/scripts/update_ginkgo_header.sh
+find . -type f \( -name '*.cuh' -o -name '*.hpp' -o -name '*.hpp.in' -o -name '*.cpp' -o -name '*.cu' -o -name '*.hpp.inc' \) -exec clang-format-8 -i "{}" \;
```

Comment:
Isn't clang-format clang-format-8?
If it is not, I think format_header use different clang-format.

Reply:
I put the comment in wrong place. This comment is for format.sh

Reply:
I'm not sure, I wanted to make sure explicitly :)
I also decided to leave format_header out for now, since it takes much longer and is possibly less stable than clang-format.
Or do you think it can be safely run for every header?

Reply:
Maybe add another comment `format_header!` to execute it separately? we can also review the change from `format_header` clearly. 
we can only run the updated files in the branch not all files in repo in format related behavior.